homeassistant/components/watergate/coordinator.py

```diff
@@ -27,9 +49,31 @@ def __init__(self, hass: HomeAssistant, api: WatergateLocalApiClient) -> None:
         )
         self.api = api
 
-    async def _async_update_data(self) -> DeviceState:
+    async def _async_update_data(self) -> WatergateAgregatedRequests:
+        # Question: Currently we are assuming that to make sure that whole device works we should have all data fetched.
+        # Is it a good approach? Should we fetch data separately?
+        # When we will change it to fetch data separately, we should also move that entity unavailaibility mechanism to entities, right?
         try:
             state = await self.api.async_get_device_state()
+            telemetry = await self.api.async_get_telemetry_data()
+            networking = await self.api.async_get_networking()
+            auto_shut_off_report = await self.api.async_get_auto_shut_off_report()
+            return WatergateAgregatedRequests(
+                state,
+                telemetry,
+                networking,
+                auto_shut_off_report,
+            )
         except WatergateApiException as exc:
-            raise UpdateFailed from exc
-        return state
+            raise UpdateFailed(f"Sonic device is unavailable: {exc}") from exc
+
+    def async_set_updated_data(self, data: WatergateAgregatedRequests) -> None:
+        """Manually update data, notify listeners and DO NOT reset refresh interval."""
+
+        self.data = data
+        self.logger.debug(
+            "Manually updated %s data",
+            self.name,
+        )
+
+        self.async_update_listeners()
```

Comment:
I think this already is a default implementation.

Maybe it would be a nice pattern that the webhook just pushes the data to the coordinator and the coordinator decides what to do with that data

Reply:
The change from the default implementation is that I am not changing refresh interval - This is why it is custom method.

Reply:
Please add a comment to highlight that accordingly

Reply:
It is written in method description. Should I add it more explicitly?